charts/redpanda/servicemonitor.go

```diff
@@ -0,0 +1,74 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// +gotohelm:filename=servicemonitor.go.tpl
+package redpanda
+
+import (
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/redpanda-data/helm-charts/pkg/gotohelm/helmette"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func ServiceMonitor(dot *helmette.Dot) *monitoringv1.ServiceMonitor {
+	values := helmette.Unwrap[Values](dot.Values)
+	
+	if !values.Monitoring.Enabled {
+		return nil
+	}
+
+	monitorLabels := helmette.Merge(FullLabels(dot), values.Monitoring.Labels)
+
+	matchLabels := map[string]string{
+		"monitoring.redpanda.com/enabled": "true",
+		"app.kubernetes.io/name":          Name(dot),
+		"app.kubernetes.io/instance":      dot.Release.Name,
+	}
+
+	tlsConfig := values.Monitoring.TLSConfig
+
+	if tlsConfig != nil {
+		tlsConfig.SafeTLSConfig.InsecureSkipVerify = false
```

Comment:
This line is very confusing to me. The default of `InsecureSkipVerify` is false, why are we explicitly setting it here? Shouldn't we just accept the default or whatever the user has provided?

Reply:
Ah, I didn't realize this would be keeping up the existing behavior. We can leave it as is for now. Can you add a note that this was pre-existing behavior and is likely less than ideal?